src/workerd/api/queue.c++

```diff
@@ -394,23 +390,29 @@ void QueueMessage::retry() {
     IoContext::current().logWarning(msg);
     return;
   }
-  result->explicitRetries.findOrCreate(id, [this]() { return kj::heapString(id); } );
+
+  result->retries.upsert(kj::heapString(id), {});
+  KJ_IF_SOME(opts, options) {
+    KJ_IF_SOME(secs, opts.delaySeconds) {
+      result->retries.upsert(kj::heapString(id), {.delaySeconds = secs});
```

Comment:
There's no need to allocate another copy of `id` and do an upsert here. The `upsert` call above returns a reference to the entry, so this code would be more efficient (and not really any more complicated) as something like:

```
  auto& entry = result->retries.upsert(kj::heapString(id), {});
  KJ_IF_SOME(opts, options) {
    KJ_IF_SOME(secs, opts.delaySeconds) {
      entry.value.delaySeconds = secs;
    }
  }
```